code/FulpstationCode/fulp_medborg/fulp_medborg_updates.dm

```diff
@@ -0,0 +1,105 @@
+		R.module.basic_modules += MB
+		R.module.add_module(MB, FALSE, TRUE)
+
+/obj/item/borg/upgrade/medbeam/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/gun/medbeam/cyborg/MB = locate() in R.module
+		R.module.remove_module(MB, TRUE)
+
+/obj/item/gun/medbeam/cyborg
+	name = "Integrated Medical Beamgun"
+	desc = "Advanced protonic nano-something or other miracle healing beam. Crossing its stream with another is ill-advised."
+	var/power_cost = 75
+
+/obj/item/gun/medbeam/cyborg/process_fire(atom/target, mob/living/silicon/robot/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(user.cell.charge < power_cost) //Check if we have enough power.
+		to_chat(user, "<span class='warning'>Power inadequate to initiate beam projection.</span>")
+		LoseTarget()
+		return
+
+	playsound(target, 'sound/effects/magic.ogg', 10, TRUE, TRUE)
+	. = ..()
+
+/obj/item/gun/medbeam/cyborg/on_beam_tick(mob/living/target)
+	if(!istype(loc,/mob/living/silicon/robot))
+		return
+
+	var/mob/living/silicon/robot/user = loc
+
+	if(!user || !target) //Sanity
+		LoseTarget()
+		to_chat(user, "<span class='warning'>No target or user detected. Aborting.</span>")
+		return
+
+	if(user.cell.charge < power_cost) //Check if we have enough power.
+		to_chat(user, "<span class='warning'>Power inadequate to maintain beam projection.</span>")
+		LoseTarget()
+		return
+
+	user.cell.charge -= power_cost
+
+	playsound(target, 'sound/effects/magic.ogg', 10, TRUE, TRUE)
+
+	. = ..()
+
+/obj/item/gun/medbeam/cyborg/attack_self(mob/living/silicon/robot/user)
+	. = ..()
+	LoseTarget()
+	playsound(get_turf(src), 'sound/machines/chime.ogg', 30, TRUE)
+	to_chat(user, "<span class='warning'>You deactivate the beam projector.</span>")
+	return
+
+
+/obj/item/borg/apparatus/beaker/medical //Medborgs can now manipulate medicines and medicine containers
+	name = "medicine storage apparatus"
+	desc = "A special apparatus for carrying beakers, medicines and medicine containers without spilling the contents. Alt-Z or right-click to drop its contents."
+	storable = list(/obj/item/reagent_containers/glass/beaker,
+				/obj/item/reagent_containers/glass/bottle,
+				/obj/item/reagent_containers/pill/patch,
+				/obj/item/reagent_containers/medigel,
+				/obj/item/reagent_containers/pill,
+				/obj/item/reagent_containers/chem_pack,
+				/obj/item/reagent_containers/hypospray/medipen,
+				/obj/item/reagent_containers/blood)
+
+/obj/item/borg/apparatus/beaker/medical/extra
+	name = "secondary medicine storage apparatus"
+	desc = "A special apparatus for carrying beakers, medicines and medicine containers without spilling the contents. Alt-Z or right-click to drop its contents."
+
+/mob/living/silicon/robot/proc/fulp_borg_unbuckle(mob/living/M) //Allows borgs to unbuckle people via loading them.
+	if(!M.buckled)
+		return
+	if(M.buckled)
+		M.buckled.unbuckle_mob(M, force = TRUE)
+	if(M.has_buckled_mobs())

```

Comment:
note that, because of the return statement earlier, if M ISN'T buckled to something, then the mobs that are buckled to M WON'T be unbuckled from them

Reply:
You mean if mobs other than M are buckled to the target object, those mobs won't be unbuckled? That's fine. It's a targeted unbuckle.

Reply:
no, I mean, if mobs are buckled to *M*, then when M is unbuckled, they won't be unbuckled from M

Reply:
Working as intended: The point of this proc is to only alter the buckle state of M and all things buckled to M if it is indeed buckled. If it is not buckled, we don't do shit.